backend/src/main/java/com/elshop/backend/product/ProductService.java

```diff
@@ -49,4 +49,13 @@ public ProductsListResponse getAll(int page, int size) {
                 pageProduct.getContent()
         );
     }
+
+
+    public void deleteProduct(String id) {
+        Optional<Product> optionalProduct = productRepository.findById(id);
+        if (optionalProduct.isEmpty()) {
+            throw new ResourceNotFoundException("Product", id);
```

Comment:
Shouldn't this exception be caught somewhere in your controller?

Reply:
It caught globally by  AppControllerAdvisor and there I handle exception properly.